src/bors/handlers/pr_events.rs

```diff
@@ -53,6 +54,13 @@ pub(super) async fn handle_push_to_pull_request(
 
     mergeable_queue.enqueue(repo_state.repository().clone(), pr_number);
 
+    if let Some(auto_build) = &pr_model.auto_build {
+        if auto_build.status == BuildStatus::Pending {
+            tracing::info!("Cancelling auto build for PR {pr_number} due to push");
+            cancel_build_workflows(&repo_state.client, &db, auto_build).await?;
```

Comment:
We should post a comment on the PR, otherwise it would be quite weird why was a running auto build cancelled out of nowhere.